MiniShopping/MiniShopping/Controllers/ProductsViewController.m

```diff
@@ -0,0 +1,86 @@
+#import "ProductsViewController.h"
+#import "ProductsView.h"
+#import "ProductCell.h"
+#import "Product.h"
+#import "FetchProducts.h"
+#import "NSMutableArray+Sorting.h"
+
+
+@interface ProductsViewController () <UICollectionViewDelegate, UICollectionViewDataSource, FetchProductsDelegate, ProductViewDelegate> {
+    NSMutableArray<Product *> *products;
+    ProductsView *productsView;
+    FetchProducts *fetchProducts;
+}
+
+@end
+
+@implementation ProductsViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.title = @"Mini Shopping";
+    self.view.backgroundColor = UIColor.systemBackgroundColor;
+    [self setupModel];
+    [self setupView];
+}
+
+- (void)setupModel {
+    products = [[NSMutableArray alloc] init];
+    fetchProducts = [[FetchProducts alloc] init];
+    [fetchProducts fetchData];
+    fetchProducts.delegate = self;
+}
+
+- (void)setupView {
+    productsView = [[ProductsView alloc] initWithFrame:CGRectZero];
+    productsView.delegate = self;
+    [productsView.productsGridView registerClass:[ProductCell class] forCellWithReuseIdentifier:@"productsGridViewIdentifier"];
+    productsView.productsGridView.translatesAutoresizingMaskIntoConstraints = NO;
```

Comment:
Nit:  For productsView, `translatesAutoresizingMaskIntoConstraints ` should be set to `NO`.

```suggestion
    productsView.translatesAutoresizingMaskIntoConstraints = NO;
```